crates/topos-core/src/types.rs

```diff
@@ -0,0 +1,105 @@
+use topos_uci::{Certificate, CertificateId};
+
+use serde::{Deserialize, Serialize};
+
+use self::stream::CertificateSourceStreamPosition;
+use topos_api::grpc::{
+    checkpoints::SourceStreamPosition,
+    tce::v1::{ProofOfDelivery as GrpcProofOfDelivery, SignedReady},
+    ConversionError,
+};
+
+pub mod stream;
+
+pub type Ready = String;
+pub type Signature = String;
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub struct CertificateDelivered {
+    pub certificate: Certificate,
+    pub proof_of_delivery: ProofOfDelivery,
+}
+
+/// Certificate's Proof of Delivery
+///
+/// This structure is used to prove that a certificate has been delivered.
+/// It contains the certificate's ID, the position of the certificate in the
+/// source stream, the list of Ready messages received and the threshold.
+/// The threshold is the number of Ready messages required to consider the
+/// certificate as delivered. For one certificate, multiple Proof of Delivery
+/// can be created on the network, each one with a different list of Ready messages.
+///
+/// Two different Proof of Delivery can still be valid if their Ready messages are
```

Comment:
```suggestion
/// Two different Proofs of Delivery can still be valid if their Ready messages are
```
Not sure I understand this right; should it be "two different PoDs for the **same** Certificate can still be valid … …"?